### PR TITLE
DNN-20026: moving necessary files from Build repository …

### PR DESCRIPTION
**Pull request by galatrash**
…so generating artifacts is all contained in this repository without dependencies.

Added some batch jobs for CLI builds and POC. In CI/CD I will use the '.build' files directly from the build jobs in Jenkins.

### CODE REVIEW

**Comment by bdukes** [comment]
I ran `Build_Source_Package.bat` and received this error output from MSBuild

```
Microsoft (R) Build Engine version 14.0.25420.1
Copyright (C) Microsoft Corporation. All rights reserved.

Build started 5/8/2018 1:02:17 PM.
Project "C:\Code\Dnn.Platform\Platform\Build\BuildScripts\CreateCommunityPackages.build" on node 1 (CreateSource target(s)).
VersionCheck:
  SolutionInfo: C:\Code\Dnn.Platform\Platform\Build\BuildScripts\..\..\SolutionInfo.cs
C:\Code\Dnn.Platform\Platform\Build\BuildScripts\Variables.build(100,7): error MSB4062: The "DotNetNuke.MSBuild.Tasks.GetVersion" task could no t be loaded from the assembly C:\Program Files (x86)\MSBuild\DotNetNuke.MSBuild.Tasks\DotNetNuke.MSBuild.Tasks.dll. Could not load file or asse mbly 'file:///C:\Program Files (x86)\MSBuild\DotNetNuke.MSBuild.Tasks\DotNetNuke.MSBuild.Tasks.dll' or one of its dependencies. The system cann ot find the file specified. Confirm that the <UsingTask> declaration is correct, that the assembly and all its dependencies are available, and
that the task contains a public class that implements Microsoft.Build.Framework.ITask. [C:\Code\Dnn.Platform\Platform\Build\BuildScripts\Create CommunityPackages.build]
Done Building Project "C:\Code\Dnn.Platform\Platform\Build\BuildScripts\CreateCommunityPackages.build" (CreateSource target(s)) -- FAILED.


Build FAILED.

"C:\Code\Dnn.Platform\Platform\Build\BuildScripts\CreateCommunityPackages.build" (CreateSource target) (1) ->
(VersionCheck target) ->
  C:\Code\Dnn.Platform\Platform\Build\BuildScripts\Variables.build(100,7): error MSB4062: The "DotNetNuke.MSBuild.Tasks.GetVersion" task could
not be loaded from the assembly C:\Program Files (x86)\MSBuild\DotNetNuke.MSBuild.Tasks\DotNetNuke.MSBuild.Tasks.dll. Could not load file or as sembly 'file:///C:\Program Files (x86)\MSBuild\DotNetNuke.MSBuild.Tasks\DotNetNuke.MSBuild.Tasks.dll' or one of its dependencies. The system ca nnot find the file specified. Confirm that the <UsingTask> declaration is correct, that the assembly and all its dependencies are available, an d that the task contains a public class that implements Microsoft.Build.Framework.ITask. [C:\Code\Dnn.Platform\Platform\Build\BuildScripts\Crea teCommunityPackages.build]

    0 Warning(s)
    1 Error(s)

Time Elapsed 00:01:22.06
```

**Comment by galatrash** [comment]
@bdukes
In my computer I have the DotNetNuke build tasks installed under `C:\Program Files (x86)\MSBuild\DotNetNuke.MSBuild.Tasks` which may be why it worked for me and not for you.  Please try again to see if the problem is resolved after adding some missing files.

**Comment by tpluscode** [comment]
That's far from ideal. Either let's commit those build targets to the repository or distribute them in a NuGet package

**Comment by galatrash** [comment]
@tpluscode I can't add these as nuget packages as they are different for different products. I'm committing them to the DNN.Platform repo.

@bdukes I found a newer DLL which I updated and pushed. Please try again.

**Comment by bdukes** [comment]
@galatrash that worked (along with the change in `Variables.build`

```
diff --git i/Build/BuildScripts/Variables.build w/Build/BuildScripts/Variables.build
index bffe5857d..c3dc94ab3 100644
--- i/Build/BuildScripts/Variables.build
+++ w/Build/BuildScripts/Variables.build
@@ -3,12 +3,12 @@
   <Import Project="AT.MSBuild.Tasks.Targets" />
   <PropertyGroup>
     <DotNetNukeMSBuildTasksPath Condition="'$(DotNetNukeMSBuildTasksPath)' == ''">$(MSBuildExtensionsPath)\DotNetNuke.MSBuild.Tasks</DotNetNukeMSBuildTasksPath>
-    <DotNetNukeMSBuildTasksLib>$(DotNetNukeMSBuildTasksPath)\DotNetNuke.MSBuild.Tasks.dll</DotNetNukeMSBuildTasksLib>
+    <DotNetNukeMSBuildTasksLib>DotNetNuke.MSBuild.Tasks.dll</DotNetNukeMSBuildTasksLib>
   </PropertyGroup>
   <UsingTask AssemblyFile="$(DotNetNukeMSBuildTasksLib)" TaskName="DotNetNuke.MSBuild.Tasks.GenerateSolutionFiles" />
   <UsingTask AssemblyFile="$(DotNetNukeMSBuildTasksLib)" TaskName="DotNetNuke.MSBuild.Tasks.ProcessSolutionFiles" />
```

**Comment by bdukes** [comment]
So, this source zip only includes the source from the _Dnn.Platform_ repo, is there going to be a further script that will pull in source from other repos (e.g. Persona Bar)?

**Comment by galatrash** [comment]
@bdukes please try with my latest commits.

**Comment by bdukes** [comment]
@galatrash we're back to the error (i.e. still looking in `Program Files` for `DotNetNuke.MSBuild.Tasks.dll`)

```
❯ .\Build_Source_Package.bat
.
.-x-x-x-x-x-x-x-x-x-x-x-x-x-x-x-x-x-x-x-x-x-x-x-x-x-x-x-x-x-x-x-x-
.
. Creating source package using:
.    "C:\Program Files (x86)\MSBuild\14.0\Bin\MSBuild.exe"
.
.-x-x-x-x-x-x-x-x-x-x-x-x-x-x-x-x-x-x-x-x-x-x-x-x-x-x-x-x-x-x-x-x-
.
Microsoft (R) Build Engine version 14.0.25420.1
Copyright (C) Microsoft Corporation. All rights reserved.

Build started 5/8/2018 3:51:25 PM.
Project "C:\Code\Dnn.Platform\Platform\Build\BuildScripts\CreateCommunityPackages.build" on node 1 (CreateSource target(s)).
VersionCheck:
  SolutionInfo: C:\Code\Dnn.Platform\Platform\Build\BuildScripts\..\..\SolutionInfo.cs
C:\Code\Dnn.Platform\Platform\Build\BuildScripts\Variables.build(100,7): error MSB4062: The "DotNetNuke.MSBuild.Tasks.GetVersion" task could no t be loaded from the assembly C:\Program Files (x86)\MSBuild\DotNetNuke.MSBuild.Tasks.dll. Could not load file or assembly 'file:///C:\Program
Files (x86)\MSBuild\DotNetNuke.MSBuild.Tasks.dll' or one of its dependencies. The system cannot find the file specified. Confirm that the <Usin gTask> declaration is correct, that the assembly and all its dependencies are available, and that the task contains a public class that impleme nts Microsoft.Build.Framework.ITask. [C:\Code\Dnn.Platform\Platform\Build\BuildScripts\CreateCommunityPackages.build]
Done Building Project "C:\Code\Dnn.Platform\Platform\Build\BuildScripts\CreateCommunityPackages.build" (CreateSource target(s)) -- FAILED.


Build FAILED.

"C:\Code\Dnn.Platform\Platform\Build\BuildScripts\CreateCommunityPackages.build" (CreateSource target) (1) ->
(VersionCheck target) ->
  C:\Code\Dnn.Platform\Platform\Build\BuildScripts\Variables.build(100,7): error MSB4062: The "DotNetNuke.MSBuild.Tasks.GetVersion" task could
not be loaded from the assembly C:\Program Files (x86)\MSBuild\DotNetNuke.MSBuild.Tasks.dll. Could not load file or assembly 'file:///C:\Progra m Files (x86)\MSBuild\DotNetNuke.MSBuild.Tasks.dll' or one of its dependencies. The system cannot find the file specified. Confirm that the <Us ingTask> declaration is correct, that the assembly and all its dependencies are available, and that the task contains a public class that imple ments Microsoft.Build.Framework.ITask. [C:\Code\Dnn.Platform\Platform\Build\BuildScripts\CreateCommunityPackages.build]

    0 Warning(s)
    1 Error(s)

Time Elapsed 00:01:08.96
Press any key to continue . . .
```

**Comment by galatrash** [comment]
@bdukes I fixed as per your initial diff. It seems the `MSBuildExtensionsPath` is referring to VS's build tasks folder.

As for including other packages in the source; e.g., Admin Experience, is out of scope for this task and is open as community effort to enhance upon these build scripts. Note these scripts are legacy ones and might contain many unused features and require some clean up before enhancement. My task was to make the base builds all inclusive in each repository without needing a special build repository.

**Comment by bdukes** [comment]
That worked for me @galatrash.

Thanks for the clarification on the scope of the task, that helps us know where to go from here.

File: Build_Upgrade_Package.bat [hunk 1]
```diff
@@ -30,15 +30,26 @@ goto finish
...
-"%builder%" /t:CreateUpgrade /v:m Build/BuildScripts/DNN_Package.build
+pushd Artifacts
+del /f /q *Upgrade.zip
+popd
+pushd "DNN Platform"
+git clean -xdf
+popd
+pushd Website
+git clean -xdf
+popd
```
Comment (tpluscode) [inline Build_Upgrade_Package.bat]: I don't think I'm a fan of git-cleaning every single time. It takes quite a while and usually should not be necessary.

Applies to all scripts ofc

Comment (galatrash) [reply Build_Upgrade_Package.bat]: Without cleaning, you will not receive clean artifacts and the produced zip files will contain some unnecessary files in them, especially the source code.

**Pull request merged**